payloadhandler.py

```diff
@@ -0,0 +1,199 @@
+import random
+import re
+from copy import copy
+from errorlogparser import ServoErrorLogParser
+
+
+class PayloadHandler():
+    def handle_payload(self, payload):
+        raise NotImplementedError
+
+
+class TravisPayloadHandler(PayloadHandler):
+    msg_template = "Please fix the following error:\n\n**File:** {}\n**Line Number:** {}\n**Error:** {}"
+
+    def __init__(self, github, travis, error_parser):
+        self.travis = travis
+        self.github = github
+        self.error_parser = error_parser
+
+    def handle_payload(self, payload):
+        commit_id = self._get_commit_id(payload)
+        build_data = self.travis.get_build(self._get_build_id(payload))
+        log = self.travis.get_log(build_data)
+        err_data = self.error_parser.parse_log(log)
+        pr_num = self.travis.get_pull_request_number(build_data)
+
+        existing_comments = [self._build_existing_comment_dict(comment) for comment in self.github.get_review_comments(pr_num)]
+        new_comments = [self._build_review_comment_dict(err_datum) for err_datum in err_data]
+
+        comments_to_post = self._delete_existing_comments(new_comments, existing_comments)
+
+        for comment in comments_to_post:
+            self.github.post_review_comment(pr_num, commit_id, comment[self.error_parser.body_key], comment[self.error_parser.path_key], comment[self.error_parser.position_key])
+
+    def _build_review_comment_dict(self, err_datum):
+        new_datum = err_datum.copy()
+        new_datum[self.error_parser.body_key] = self.msg_template.format(err_datum[self.error_parser.path_key], err_datum[self.error_parser.position_key], err_datum[self.error_parser.body_key])
+
+        return new_datum
+
+    def _build_existing_comment_dict(self, comment):
+        new_comment = {}
+
+        new_comment[self.error_parser.body_key] = comment[self.error_parser.body_key]
+        new_comment[self.error_parser.path_key] = comment[self.error_parser.path_key]
+        new_comment[self.error_parser.position_key] = comment[self.error_parser.position_key]
```

Comment:
I'm not clear on the purpose of this - the resulting dictionary is identical to the source, right? I feel like this ends up appearing more complicated than it needs to be.